tools/PI/DevHome.PI/Helpers/WindowHelper.cs

```diff
@@ -632,4 +628,83 @@ internal static async void ShowTimedMessageDialog(FrameworkElement frameworkElem
             _log.Error(ex, "Error showing timed message dialog");
         }
     }
+
+    internal enum CpuArchitecture
+    {
+        X86,
+        X64,
+        ARM,
+        ARM64,
+        Unknown,
+    }
+
+    internal static CpuArchitecture GetTargetArchitecture(IMAGE_FILE_MACHINE target)
+    {
+        return target switch
+        {
+            IMAGE_FILE_MACHINE.IMAGE_FILE_MACHINE_I386 => CpuArchitecture.X86,
+            IMAGE_FILE_MACHINE.IMAGE_FILE_MACHINE_AMD64 => CpuArchitecture.X64,
+            IMAGE_FILE_MACHINE.IMAGE_FILE_MACHINE_ARM => CpuArchitecture.ARM,
+            IMAGE_FILE_MACHINE.IMAGE_FILE_MACHINE_ARM64 => CpuArchitecture.ARM64,
+            _ => CpuArchitecture.Unknown,
+        };
+    }
+
+    internal static string GetAppArchitecture(SafeProcessHandle handle, string moduleName)
+    {
+        var cpuArchitecture = CommonHelper.GetLocalizedString("CpuArchitecture_Unknown");
+        unsafe
+        {
+            IMAGE_FILE_MACHINE processInfo;
+            IMAGE_FILE_MACHINE machineInfo;
+            var isWow64Result = PInvoke.IsWow64Process2(handle, out processInfo, &machineInfo);
+            if (isWow64Result)

```

Comment:
NIT: Prefer early return over deeply nested if's.

Reply:
Agreed, although in this case we need the deeply-nested ifs because we have 5 possible return cases, and they're dependent on the earlier conditions.